### PR TITLE
[infra] Add jsoncpp library

### PR DESCRIPTION
**Pull request by jinevening**
This adds jsoncpp library to the project.

ONE-DCO-1.0-Signed-off-by: Hyukjin Jeong <hj1.jeong@samsung.com>

---
Related to: https://github.com/Samsung/ONE/issues/8399
Draft PR: #8460

### CODE REVIEW

File: infra/cmake/packages/JsoncppConfig.cmake [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+  endif(NOT JsoncppSource_FOUND)
+
+  nnas_include(ExternalBuildTools)
+  ExternalBuild_CMake(CMAKE_DIR   ${JsoncppSource_DIR}
+                      BUILD_DIR   ${CMAKE_BINARY_DIR}/externals/JSONCPP/build
+                      INSTALL_DIR ${EXT_OVERLAY_DIR}
+                      IDENTIFIER  "1.9.5"
+                      PKG_NAME    "JSONCPP"
+                      EXTRA_OPTS "-DBUILD_STATIC_LIBS=ON"
+                                 "-DBUILD_SHARED_LIBS=OFF")
```
Comment (jinevening) [inline infra/cmake/packages/JsoncppConfig.cmake:16]: This is for using static library

**Review by mhs4670go — APPROVED** [review]
LGTM

File: infra/cmake/packages/JsoncppConfig.cmake (hunk 1 above)
Comment (chunseoklee) [reply infra/cmake/packages/JsoncppConfig.cmake:16]: Is there any reason not to use so binary ? FYI, tizen's jsoncpp-devel package provides so files.

Comment (jinevening) [reply infra/cmake/packages/JsoncppConfig.cmake:16]: It is for ease of release. I think that using system-installed libraries (ex: jsoncpp-devel in tizen) will make release difficult.

Comment (chunseoklee) [reply infra/cmake/packages/JsoncppConfig.cmake:16]: Ah static link is better for offline tool. Lets discuss later about tizen case

**Review by chunseoklee — APPROVED** [review]

**Review by seanshpark — APPROVED** [review]
LGTM

**Comment by hseok-oh** [comment]
FYI

1) Runtime is using `jsoncpp` on `runtime/3rdparty/jsoncpp`
2) Tizen is using `jsoncpp` 1.9.2

**Review by hseok-oh — APPROVED** [review]
LGTM

**Comment by chunseoklee** [comment]
> Tizen is using jsoncpp 1.9.2

Recently, tizen bump to 1.9.5

**Pull request merged**